ERCS/erc-8006.md

```diff
@@ -0,0 +1,370 @@
+
+## Motivation
+
+For many smart contract software systems, compliance – and the validations it relies on – is an essential and often mandatory component. From the simplest vaults to decentralised exchanges, oracles, and smart accounts, all require **restrictions**, **requirements**, **prohibitions**, and **custom validation rules** – for example, limiting administrative privileges, restricting access to public methods, capping withdrawal amounts, and more – with the ability to modify these rules when necessary, especially considering ongoing regulatory developments across the globe.
+
+### Current State of Smart Contract Rule Implementation Capabilities
+
+Current smart contract approaches allow for the creation of simple validation rules through algorithmic constraints and modifiers. However, more complex rules – such as dynamic, composite, or conditional ones – while still possible, become increasingly difficult to implement as the number of inputs and conditions grows over time.
+
+Moreover, certain problems, such as interactive composition of simple rules, cannot be addressed with current methods due to the lack of reflection capabilities in smart contract platforms. Reusing complex rules, modifying them dynamically, and hierarchical organization are similarly challenging.
+
+### Compliance Problems Faced by Developers
+
+When implementing compliance conditions, developers face several important challenges:
+
+1. **Code Complexity**: Beyond simple value thresholds, compliance logic becomes increasingly complex. This complexity makes codebases significantly harder to maintain, understand, and audit effectively. The intricate nature of compliance requirements often results in convoluted implementations that obscure the underlying business logic.
+
+2. **Security Vulnerabilities**: Flaws in rule implementation pose serious risks to system integrity and user assets. Poor implementations can enable unauthorized access to restricted functionality, lead to direct loss of user funds through exploits, and potentially violate regulatory requirements that the rules were meant to enforce.
+
+3. **Integration Difficulties**: Adding new compliance logic to existing codebases often requires:
+   - Extensive codebase changes
+   - Separate integration work for each compliance rule implementation
+   - The need of carefully conducted contract upgrade process (which is error-prone)
+
+4. **Redundant Implementation**: Many compliance patterns appear across different projects but lack standard interfaces, forcing developers to repeatedly implement similar functionality.
+
+5. **Costly Updates**: As regulatory requirements change, contracts need updates, requiring the entire development process again - writing, testing, auditing, and deploying - increasing costs and error potential.
```

Comment:
This lays out the problem space quite nicely, but you should also use the Motivation to explain how your particular proposal is the best solution to that problem. Why should an implementer choose your solution over a (possibly hypothetical) alternative ERC?